### ISSUE TITLE
Cannot declare Array type as a type hint of a SerializerMethodField

### ISSUE THREAD

**Issue by thomasWajs** [axnsan12/drf-yasg#584]
Hi, I'm beggining to use this project to generate the documentation of our API and it's really useful, thanks !

I'm trying to use the [type hinting on a serializer method](https://drf-yasg.readthedocs.io/en/stable/custom_spec.html?#support-for-serializermethodfield) to declare the return type of my method as an array of objects :

```
class MySerializer(serializers.Serializer):
    field = serializers.SerializerMethodField()
    
    def get_field(self, instance) -> typing.List[dict]:
        return [
            {'key': 'value'}
        ]
```

But this generate

```
type: "object"
```

instead of 

```
type: "array"
items:
    type: "object"
```

I've traced down the issue to this line : https://github.com/axnsan12/drf-yasg/blob/master/src/drf_yasg/inspectors/field.py#L613 which keeps only the first args of a non-class hint.

What's strange is that there's some handling of Sequences in `inspect_collection_hint_class` but there's no way to reach it, because typing.Sequence subclasses are not classes and are discarded by the test `if inspect.isclass(hint_class)` in `SerializerMethodFieldInspector`.

Is there a way to generate my schema that I overlooked ?

Thanks

**Comment by zhaojiejoe** [comment]
https://drf-yasg.readthedocs.io/en/stable/custom_spec.html?#support-for-serializermethodfield 
I can't find the way to use "type hints" to slove the question. But the swagger_serializer_method can help you.
@swagger_serializer_method(serializer_or_field=OtherStuffSerializer(many=True))

**Comment by thomasWajs** [comment]
Ok, thanks for the confirmation it's not feasible with type hints.

I will look into the other notation.

**Comment by chgad** [comment]
I think what @zhaojiejoe mentioned is the way to go as the project does not support docs from typehinting.

**Comment by thomasWajs** [comment]
@chgad  I'm not quite sure what you're referring to, there's definitely support for type hinting on `serializers.SerializerMethodField` :

https://drf-yasg.readthedocs.io/en/stable/custom_spec.html?#support-for-serializermethodfield
> For simple cases where the method is returning one of the supported types, Python 3 type hinting of the serializer method return value can be used

**Comment by chgad** [comment]
Oh yea. You are actually right but i'm not sure if this holds for nested Typehinting as you mentioned in your example. Have you tried modifying `List[Dict]` to just `List[]` and checked if the docs are created accordingly.

**Comment by thomasWajs** [comment]
I tried every combination I could think of, but none would generate the correct result.

There is some handling of Sequences ( typing.List is a Sequence ) in `inspect_collection_hint_class`, but we never reach this part of the code because `SerializerMethodFieldInspector` will discard the Sequences.

**Comment by chgad** [comment]
Yes, but not only that i think if the code would be reached it would cause an error since i can not find the variables `typing` and `inspect_signiture` mentioned [here](https://github.com/axnsan12/drf-yasg/blob/master/src/drf_yasg/inspectors/field.py#L609) anywhere in the method.

**Comment by thomasWajs** [comment]
Lines 22 to 30 :

```
try:
    import typing
except ImportError:
    typing = None

try:
    from inspect import signature as inspect_signature
except ImportError:
    inspect_signature = None
```

**Comment by chgad** [comment]
Oops. Seems like i was a bit too fast with my reasoning there. But nonethe less you are right. The whole typehinting stuff seems to be skipped/discarded whatsoever.

**Comment by thomasWajs** [comment]
It does work on hints made with classes :

```
def get_field(self, instance) -> int:
```

will yield the correct result.

As said in my first message, the incorrect behaviour appears with non-class hints, because of https://github.com/axnsan12/drf-yasg/blob/master/src/drf_yasg/inspectors/field.py#L613
